src/components/main-view/main-view.jsx

```diff
@@ -2,63 +2,117 @@ import React, { useState, useEffect } from 'react';
 import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
 import { LoginPage } from "../login-view/login-page";
-import "./main-view.css"
+import { SignUp } from '../signup-view/signup-view';
+import "./main-view.css";
+import { Button, Row, Col } from "react-bootstrap";
 
 export const MainView = () => {
-  const [movies, setMovies] = useState([]);
 
+  const storedToken = localStorage.getItem("token");
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const [movie, setMovie] = useState([]);
   // state changes for selected movies
   const [selectedMovie, setSelectedMovie] = useState(null);
-  const [user, setUser] = useState(null);
+  const [user, setUser] = useState(storedUser ? storedUser : null);
+  const [token, setToken] = useState(storedToken ? storedToken : null);
+
+  const chunkArray = (array, chunkSize) => {
+    let results = [];
+    while (array.length) {
+      results.push(array.splice(0, chunkSize));
+    }
+    return results;
+  };
+
+  const movieRows = chunkArray([...movie], Math.ceil(movie.length / 2));
 
   useEffect(() => {
-    fetch('https://movie-api-uahq.onrender.com/movies')
+    if (!token) return;
+    fetch('https://movie-api-wbl0.onrender.com/movies', {
+      headers: { Authorization: `Bearer ${token}` },
+    })
+
       .then((response) => response.json())
       .then((data) => {
-        console.log(data);
-        setMovies(data);
+        console.log("movies from api:", data);
+        const moviesFromApi = data.map((movie) => {
+          return {
+            _id: movie._id,
+            Title: movie.Title,
+            Director: {
+              Name: movie.Director.Name,
+              Bio: movie.Director.Bio
+            },
+            Description: movie.Description,
+            Genre: {
+              Name: movie.Genre.Name,
+              Description: movie.Genre.Description
+            },
+            imagePath: movie.ImagePath,
+          };
+        });
```

Comment:
Unecessary, the movie already have the structure that you're trying to build using `.map`. I would just call `setMovies(data)`

Reply:
resolved